GSA-TTS/FAC
# Search Results - Summary Report Generation

# Search Results - Summary Report Generation

Best Fit Issue: https://github.com/GSA-TTS/FAC/issues/2924

## Changes:
1. Summary report generation for search results.
    1. There's a button, and it resubmits the form to a new URL & view.
    2. The new view re-runs the search and uses the results to build the summary spreadsheet. Then the user is given the file.
    3. The user clicks the button and receives a file. There's no redirection, so it may seem as though nothing is happening for a few seconds.
    4. The summary report generation is not terribly fast, so it's limited to 1000 results for now. That may still be too high.
2. Minor change for summary report generation efficiency (less total queries). 
3. Summary reports come with a coversheet note about the limit and a link to the API, in the same style as the pre-dissem sheets. 
4. Tests on the `MultipleSummaryReportDownloadView`
    * It mocks the file upload and download portions. 
    * It doesn't test the actual file, just some of the expected outputs of the view.
6. A lot of comments, formatting, and refactoring of the search view.
    * The call of `search_general` and the form cleanup have moved into their own functions.

The record limit on downloads comes from `settings.SUMMARY_REPORT_DOWNLOAD_LIMIT`. That value is used in the views and HTML. 

## How to test:
1. Switch to this branch and run normally.
2. Ensure you have data in the dissemination tables.
    * There are sql dumps available on GDrive.
3. Make a search. Verify the "Export to XLSX" button looks right.
7. Verify the "Export to XLSX" button works.
    * Small result sets should come back in under a few seconds.
    * Large result sets are capped at 1000 records.
    * A max size result set takes about 15 seconds to generate on my machine. 

## Screenshot:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/ab65d248-4807-4293-9a4d-f0cd3382bec4)

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
